src/tasks/nli_probing.py

```diff
@@ -7,123 +7,95 @@
 from .registry import register_task
 from ..utils.data_loaders import load_tsv, process_sentence
 
+@register_task('nps', rel_path='nps/')
+class NPSTask(PairClassificationTask):
+
+    def __init__(self, path, max_seq_len, name, **kw):
+        super(NPSTask, self).__init__(name, n_classes=3, **kw)
+        self.load_data(path, max_seq_len)
+        self.sentences = self.val_data_text[0] + self.val_data_text[1]
+
+    def load_data(self, path, max_seq_len):
+        targ_map = {'neutral': 0, 'entailment': 1, 'contradiction': 2}
+        prob_data = load_tsv(self._tokenizer_name, os.path.join(path, 'dev.tsv'), max_seq_len,
+                            s1_idx=0, s2_idx=1, label_idx=2, label_fn=targ_map.__getitem__, skip_rows=0)
+        self.train_data_text = self.val_data_text = self.test_data_text = prob_data
+        log.info("\tFinished loading NP/S data.")
+
 
 @register_task('nli-prob', rel_path='NLI-Prob/')
 class NLITypeProbingTask(PairClassificationTask):
     ''' Task class for Probing Task (NLI-type)'''
 
-    def __init__(self, path, max_seq_len, name, probe_path="probe_dummy.tsv",
-                 **kw):
+    def __init__(self, path, max_seq_len, name, probe_path="", **kw):
         super(NLITypeProbingTask, self).__init__(name, n_classes=3, **kw)
         self.load_data(path, max_seq_len, probe_path)
-        #  self.use_classifier = 'mnli'  # use .conf params instead
-        self.sentences = self.train_data_text[0] + self.train_data_text[1] + \
-            self.val_data_text[0] + self.val_data_text[1]
+        self.sentences = self.val_data_text[0] + self.val_data_text[1]
 
     def load_data(self, path, max_seq_len, probe_path):
         targ_map = {'neutral': 0, 'entailment': 1, 'contradiction': 2}
-        label_fn = targ_map.__getitem__
-        tr_data = load_tsv(data_file=os.path.join(path, 'train_dummy.tsv'), max_seq_len=max_seq_len,
```

Comment:
train_dummy.tsv and test_dummy.tsv are dummy. Since the tasks are only used with pretrain = 0, target_tasks_train = 0, full_eval = 1 and write_pred = val, the only split used in the program is the self.val_data_text. (this is similar to mnli-diagnostic)
What I did is just like mnli-diagnostic, pointing all splits to the only data that exists.

Reply:
There are five tasks here, they are of the same kind. So I did this five times.

Reply:
Looks good!